src/view/placeholder.js

```diff
@@ -105,8 +121,8 @@ function updateSinglePlaceholder( writer, element, info ) {
 
 	// If checkFunction is provided and returns false - remove placeholder.
 	if ( checkFunction && !checkFunction() ) {
-		if ( element.hasClass( 'ck', 'ck-placeholder' ) ) {
-			writer.removeClass( [ 'ck', 'ck-placeholder' ], element );
+		if ( element.hasClass( 'ck-placeholder' ) ) {
```

Comment:
I like that you decided to not remove `.ck` when the placeholder is not needed. It's simpler and totally reasonable – the `.ck-placeholder` class is responsible for showing the thing. The `.ck` class makes for the base.